contexts/useAddressBook.tsx

```diff
@@ -164,7 +182,7 @@ export const WithAddressBook = ({children}: {children: React.ReactElement}): Rea
 			try {
 				const existingEntry = await getEntry({address: entry.address});
 				if (existingEntry) {
-					const mergedChains = [...(existingEntry.chains || []), ...(entry.chains || [])];
+					const mergedChains = [...(entry.chains || [])];
```

Comment:
This way it's not possible to update the chains completely. It was only possible to add new chains to the entry, but not removing some of them. The reason for that is we were always spreading `existingEntry.chains` into merged chains which seemed wrong. I think that chains can be the only field to be overwritten. Can you pls provide some example when we actually need to spread the chains like this? It will help to come up with new logic for this